### DISCUSSION TITLE
Add the ability to fetch the quota (disk usage) of an account?

### DISCUSSION THREAD

**Discussion by jreyesr** [umanamente/n8n-nodes-imap#28] [Ideas]
Hello! First off, thanks for developing and sharing this node, it's really useful.

Would you be open to adding a new operation to fetch the disk usage of an account, which apparently is called the account's "quota" in IMAP? Adding that may be useful for people who want to monitor an account's remaining space, in a similar way to how the Get Status operation can already be used to monitor the amount of unread emails in a mailbox.

I've been playing arond with monitoring some mailboxes with N8N, exporting those metrics (e.g. the amount of unread emails over time) to Grafana, and I'd like to have the disk usage over time too (now I can have the count of emails, but not the size that they take, or how full the account is)

I've already cloned the plugin and quickly added support for that operation, which was fairly easy because `imapflow` [already supports that command](https://imapflow.com/module-imapflow-ImapFlow.html#getQuota):

![image](https://github.com/user-attachments/assets/b9631892-2866-41e6-8346-be5da0b84b7d)

Would you be open to integrating that new operation into this plugin, if I open a PR? I don't know this project's policy around external contributions.

Thanks for your attention!

**Comment by umanamente** [comment]
Sure, please feel free to open a PR. The project welcomes contributions

**Reply by jreyesr** [reply]
I've opened #30. Let me know if you'd like any changes to be made
